uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Import/Parser/BaseClass.php

```diff
@@ -0,0 +1,323 @@
+<?php
+
+use CRM_HRLeaveAndAbsences_BAO_LeaveRequest as LeaveRequest;
+use CRM_HRLeaveAndAbsences_BAO_AbsenceType as AbsenceType;
+
+/**
+ * Class to parse activity csv files.
+ */
+class CRM_HRLeaveAndAbsences_Import_Parser_BaseClass extends CRM_HRLeaveAndAbsences_Import_Parser {
+
+  use CRM_HRLeaveAndAbsences_Import_Helpers_DataImportTrait;
+
+  /**
+   * @var array
+   */
+  protected $_mapperKeys;
+
+  /**
+   * @var array
+   */
+  private $absenceTypes;
+
+  /**
+   * @var array
+   */
+  private $absenceStatuses;
+
+  /**
+   * @var array
+   */
+  private $leaveImportSuccess = [];
+
+  /**
+   * @var array
+   */
+  private $leaveImportError = [];
+
+  /**
+   * Class constructor.
+   *
+   * @param array $mapperKeys
+   */
+  public function __construct(&$mapperKeys) {
+    parent::__construct();
+    $this->_mapperKeys = &$mapperKeys;
+  }
+
+  /**
+   * Function of undocumented functionality required by the interface.
+   */
+  protected function fini() {}
+
+  /**
+   * The initializer code, called before the processing.
+   */
+  public function init() {
+    $activityContact = CRM_Activity_BAO_ActivityContact::import();
+    $activityTarget['target_contact_id'] = $activityContact['contact_id'];
+
+    $fields = self::getFields();
+    $fields = ['' => ['title' => ts('- do not import -')]] + $fields;
+    foreach ($fields as $name => $field) {
+      $field['type'] = CRM_Utils_Array::value('type', $field);
+      $field['dataPattern'] = CRM_Utils_Array::value('dataPattern', $field, '//');
+      $field['headerPattern'] = CRM_Utils_Array::value('headerPattern', $field, '//');
+      $this->addField($name, $field['title'], $field['type'], $field['headerPattern'], $field['dataPattern']);
+    }
+
+    $this->setActiveFields($this->_mapperKeys);
+    $this->absenceTypes = $this->getAbsenceTypes();
+    $this->absenceStatuses = array_flip(LeaveRequest::buildOptions('status_id'));
+  }
+
+  /**
+   * Handle the values in mapField mode.
+   *
+   * @param array $values
+   *   The array of values belonging to this line.
+   *
+   * @return bool
+   */
+  public function mapField(&$values) {
+    return CRM_Import_Parser::VALID;
+  }
```

Comment:
Yeah, Inherited the code from the core activity entity. I think the reason why it does nothing is because the no validation takes place during the mapping field phase of the import. Will add that to the doc block.